pytorch_lightning/trainer/trainer.py

```diff
@@ -834,6 +869,63 @@ def __test_given_model(self, model, test_dataloaders):
 
         return results
 
+    def predict(
+        self,
+        model: Optional[LightningModule] = None,
+        dataloaders: Optional[Union[DataLoader, List[DataLoader]]] = None,
+    ):
+        r"""
+
+        Separates from fit to make sure you never run on your predictions set until you want to.
+
+        This will call the model forward function to compute predictions.
+
+        Args:
+            model: The model to predict on.
+
+            dataloaders: Either a single
+                Pytorch Dataloader or a list of them, specifying inference samples.
+
+        Returns:
+            The final test result dictionary. If no test_epoch_end is defined returns a list of dictionaries
+        """
+
+        # --------------------
+        # SETUP HOOK
+        # --------------------
+        if not (
+            isinstance(dataloaders, DataLoader)
+            or isinstance(dataloaders, (list, tuple))
+            and all(isinstance(d, DataLoader) for d in dataloaders)
+        ):
+            raise MisconfigurationException(
+                'You need to pass a dataloader or a list of dataloaders to `trainer.predict`. '
+            )
+
+        if model is None:
+            raise MisconfigurationException(
+                'You need to pass a model to `trainer.predict`. '
+            )
+
+        # attach data
+        if dataloaders is not None:
+            self.data_connector.attach_dataloaders(model, test_dataloaders=dataloaders)
+
+        # set path variable
+        self.is_predicting = True
+        os.environ['PL_TESTING_MODE'] = '1'
+        self.model = model
+
+        results = self.fit(model)
```

Comment:
question: if the Trainer is set to run for N epochs, will this run for N epochs?

Reply:
Good question. It might. I will check this.